FirebaseDatabase/Sources/Core/FRepo.m

```diff
@@ -518,10 +518,8 @@ - (void)getData:(FIRDatabaseQuery *)query
         (void (^_Nonnull)(NSError *__nullable error,
                           FIRDataSnapshot *__nullable snapshot))block {
     FQuerySpec *querySpec = [query querySpec];
-    id<FNode> node =
-        [self.serverSyncTree calcCompleteEventCacheAtPath:querySpec.path
-                                          excludeWriteIds:@[]];
-    if (![node isEmpty]) {
+    id<FNode> node = [self.serverSyncTree getServerValue:[query querySpec]];
+    if (node != nil && ![node isEmpty]) {
```

Comment:
@schmidt-sebastian I'm stuck here. I can't figure out why `[node isEmpty]` is needed to pass the two tests:

https://github.com/firebase/firebase-ios-sdk/pull/7403/files#diff-3a8cb26bda8f6a6d6700b7c2ee8b1368a1c147ed7bb29743a0f6146c08d7b723R1437-R1495

The equivalent is not needed for the android changes.